### PR TITLE
chore: apply flake8-comprehensions Ruff rules to improve collection comprehensions

### PR DESCRIPTION
**Pull request by bowenliang123**
# Description

- Applied some auto-fixable Ruff rules to improve collection usages for dict, list and set:
    - C4 flake8-comprehensions: https://docs.astral.sh/ruff/rules/#flake8-comprehensions-c4
- All the code changes are fixed with Ruff by running dev/reformat script
- No major changes in logic

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [ ] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] TODO

# Suggested Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [ ] `optional` I have made corresponding changes to the documentation 
- [ ] `optional` I have added tests that prove my fix is effective or that my feature works
- [ ] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by bowenliang123** [comment]
cc @laipz8200 @crazywoola

File: dev/reformat [hunk 1]
```diff
@@ -9,7 +9,7 @@ if ! command -v ruff &> /dev/null; then
 fi
 
 # run ruff linter
-ruff check --fix ./api
+ruff check --fix --unsafe-fixes ./api
```
Comment (laipz8200) [inline dev/reformat]: Let's keep this line as it is. Removing `--unsafe-fixes` will allow poeple to check it manually, which might be better.

```suggestion
ruff check --fix ./api
```

**Review by laipz8200 — CHANGES_REQUESTED** [review]
Awesome! Please address my comments and make sure to sync with the latest code on 'main'.

File: dev/reformat (hunk 1 above)
Comment (bowenliang123) [reply dev/reformat]: Alright. SGTM.

**Review by laipz8200 — APPROVED** [review]

**Pull request merged**